File: service/src/main/java/coding/challenge/configuration/S3Properties.java
package coding.challenge.configuration;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Represents configuration properties for S3.
 *
 * @param url       S3 url
 * @param accessKey S3 access key
 * @param secretKey S3 secret key
 * @param bucket    S3 bucket
 */
@ConfigurationProperties(prefix = "s3")
public record S3Properties(String url, String accessKey, String secretKey, String bucket) {
}

File: service-kt/src/main/kotlin/coding/challenge/configuration/S3Properties.kt
package coding.challenge.configuration

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * Represents configuration properties for S3.
 *
 * @param url       S3 url
 * @param accessKey S3 access key
 * @param secretKey S3 secret key
 * @param bucket    S3 bucket
 */
@ConfigurationProperties(prefix = "s3")
data class S3Properties(val url: String, val accessKey: String, val secretKey: String, val bucket: String)
